apps/yggdrasil/test/yggdrasil/user_role_test.exs

```diff
@@ -0,0 +1,98 @@
+defmodule Yggdrasil.UserRoleTest do
+  use ExUnit.Case, async: false
+
+  alias Yggdrasil.{Repo, User, Role, UserRole, RolePermission, Resource, Permission}
+
+  @user %{username: "tester",
+          password: "password",
+          password_confirmation: "password"}
+
+  @test_perms [:foo_perm, :bar_perm, :baz_perm]
+  @test_resources [:foo_res, :bar_res, :baz_res]
+  @test_roles [
+    %{
+      name: :foo_role,
+      resources: [
+        %{
+          name: :foo_res,
+          perms: [:foo_perm, :bar_perm]
+        },
+        %{
+          name: :bar_res,
+          perms: [:bar_perm, :baz_perm]
+        }
+      ]
+    }
+  ]
+
+  setup tags do
+    unless tags[:async] do
+      Ecto.Adapters.SQL.restart_test_transaction(Yggdrasil.Repo, [])
+    end
+
+    :ok
+  end
+
+  setup _ctx do
```

Comment:
I have danced around this for now to get other things done, but I think what I want is a `UserHelper` module with functions just like you mentioned.
```elixir
setup _ctx do
  {user, roles} = UserHelper.default_user
  |> UserHelper.setup_user
  |> UserHelper.setup_roles(UserHelper.default_roles) 

  {:ok, user: user, roles: roles}
end
```
You could hide some of that, and have like a `setup_default_roles` as well etc. Just haven't had time to really dig into it. I sort of prefer the explicit nature of that.

I also think I need to start looking at tags in test, so for example the `user_test` suite doesn't seed the roles in the setup because I didn't want them to seed for all the tests in the module as it's not applicable to most of them but I believe tags can be used to drive which setup functions get applied much like it's used above to determine if async is being used.